src/components/NewTodo.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import './NewTodo.css';
+import PropTypes from 'prop-types';
+
+class NewTodo extends React.Component {
+  state = {
+    id: this.props.id,
```

Comment:
generate an id when you create a new todo, if you pass it as props and write directly to the state, you cause two additional unnecessary renders